src/routing_table/mod.rs

```diff
@@ -377,6 +398,37 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         }
     }
 
+    // Returns the collection of groups to which a peer joining our group should connect.
+    //
+    // This will generally be all the groups in our routing table.  However, if the addition of the
+    // new node will cause our group to split, only the groups which will remain neighbours after
+    // the split are returned.  Returns `Err(Error::PeerNameUnsuitable)` if `name` is not within our
+    // group.
+    // TODO - should we check `name` isn't already in RT?
+    pub fn expect_add_to_our_group(&self, name: &T) -> Result<Groups<T>, Error> {
+        if !self.our_group_prefix.matches(name) {
+            return Err(Error::PeerNameUnsuitable);
+        }
+        let mut groups = self.groups.clone();
+        let mut our_group = unwrap!(groups.remove(&self.our_group_prefix));
+        if self.should_split_our_group(&our_group) {
+            let our_prefix_after_split = Prefix::new(self.our_group_prefix.bit_count() + 1,
+                                                     self.our_name);
+            let groups_to_remove = groups.keys()
+                .filter(|&&x| {
+                    x != our_prefix_after_split && !x.is_neighbour(&our_prefix_after_split)
+                })
+                .cloned()
+                .collect_vec();
+            for prefix in groups_to_remove {
+                let _ = groups.remove(&prefix);
```

Comment:
I wonder whether we should just add those groups as empty instead. Otherwise it will probably make the invariant check in `RoutingTable::new_with_groups` fail if required neighbouring groups are missing.

Alternatively, we'd need to use `our_prefix_after_split` below - but I'm not sure whether that will cause trouble, since the other group members haven't performed the split yet?